include/audio/aseq.h

```diff
@@ -0,0 +1,1979 @@
+.endm
+
+/**
+ * font <fontId:u8>
+ *
+ *  Set the current soundfont for this channel to `fontId`.
+ */
+.macro font fontId
+    _wr_cmd_id  font, ,0xC6,,,,,,, 0, 0
+    _wr_u8      \fontId
+.endm
+
+.macro scriptctr arg
+    _wr_cmd_id  scriptctr, 0xC5,,,,,,,, 0, 0
+    _wr_u16     \arg
+.endm
+
+.macro dyntbllookup
+    _wr_cmd_id  dyntbllookup, ,0xC5,,,,,,, 0, 0
+.endm
+
+/**
+ * runseq <seqPlayer:u8> <seqId:u8>
+ *
+ *  Plays the sequence seqId on seqPlayer
+ */
+.macro runseq seqPlayer, seqId
+    _wr_cmd_id  runseq, 0xC4,,,,,,,, 0, 0
+    _wr_u8      \seqPlayer
+    _wr_u8      \seqId
+.endm
+
+#if (MML_VERSION == MML_VERSION_MM)
+
+    .macro mutechan arg0
+        _wr_cmd_id  mutechan, 0xC3,,,,,,,, 0, 0
+        _wr_s16     \arg0
+    .endm
+
+#endif
+
+/**
+ * noshort
+ *
+ *  Disable short notes encoding.
+ */
+.macro noshort
+    _wr_cmd_id  noshort, ,0xC4,,,,,,, 0, 0
+.endm
+
+/**
+ * short
+ *
+ *  Enable short notes encoding.
+ */
+.macro short
+    _wr_cmd_id  short, ,0xC3,,,,,,, 0, 0
+.endm
+
+.macro dyntbl label
+    _wr_cmd_id  dyntbl, ,0xC2,,,,,,, 0, 0
+    _wr_lbl     \label
+.endm
+
+/**
+ * instr <instNum:u8>
+ *
+ *  Set instrument `instNum` from the current soundfont as the active instrument for this channel or layer.
+ */
+.macro instr instNum
+    _wr_cmd_id  instr, ,0xC1,0xC6,,,,,, 0, 0
+    _wr_u8      \instNum
+.endm
+
+#if (MML_VERSION == MML_VERSION_MM)
+
+    .macro unk_BE arg0
+        _wr_cmd_id  unk_BE, ,0xBE,,,,,,, 0, 0
+        _wr_u8      \arg0
+    .endm
+
+#endif
+
+.macro randptr arg0, arg1
+    #if (MML_VERSION == MML_VERSION_OOT)
+        _wr_cmd_id  randptr, ,0xBD,,,,,,, 0, 0
+    #else
+        _wr_cmd_id  randptr, ,0xA8,,,,,,, 0, 0
+    #endif
+    _wr_u16         \arg0
+    _wr_u16         \arg1
+.endm
+
+#if (MML_VERSION == MML_VERSION_MM)
+
+    .macro samplestart arg0
+        _wr_cmd_id  samplestart, ,0xBD,,,,,,, 0, 0
+        _wr_u8      \arg0
+    .endm
+
+    .macro unk_AA
+        _wr_cmd_id  unk_AA, ,0xAA,,,,,,, 0, 0
+    .endm
+
+    .macro unk_A7 arg0
+        _wr_cmd_id  unk_A7, ,0xA7,,,,,,, 0, 0
+        _var        \arg0
+    .endm
+
+    .macro unk_A6 arg0, arg1
+        _wr_cmd_id  unk_A6, ,0xA6,,,,,,, 0, 0
+        _var        \arg0
+        _var        \arg1
+    .endm
+
+    .macro unk_A5
+        _wr_cmd_id  unk_A5, ,0xA5,,,,,,, 0, 0
+    .endm
+
+    .macro unk_A4 arg0
+        _wr_cmd_id  unk_A4, ,0xA4,,,,,,, 0, 0
+        _var        \arg0
+    .endm
+
+    .macro unk_A3
+        _wr_cmd_id  unk_A3, ,0xA3,,,,,,, 0, 0
+    .endm
+
+    .macro unk_A2 arg0
+        _wr_cmd_id  unk_A2, ,0xA2,,,,,,, 0, 0
+        _var        \arg0
+    .endm
+
+    .macro unk_A1
+        _wr_cmd_id  unk_A1, ,0xA1,,,,,,, 0, 0
+    .endm
+
+    .macro unk_A0 arg0
+        _wr_cmd_id  unk_A0, ,0xA0,,,,,,, 0, 0
+        _var        \arg0
+    .endm
+
+#endif
+
+.macro ptradd value
+    _wr_cmd_id  ptradd, ,0xBC,,,,,,, 0, 0
+    _wr_lbl     \value
+.endm
+
+// ptradd but for immediates instead of labels
```

Comment:
but what does ptradd do? (although I see more undocumented instructions below so I guess that's intended?)